### PR TITLE
chore: enable restricting supercharge promos

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

The needed logic is described in the linear task.

### Test plan

Updated unit tests, manual testing

### Related issues

- Fixes RET-889

### Backwards compatibility

Y

### CODE REVIEW

File: src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx [hunk 1]
```diff
@@ -297,30 +303,36 @@ export default function ConsumerIncentivesHomeScreen() {
           </Trans>
         ) : hasMaxBalance ? (
           t('superchargeDisclaimerMaxRewards', { token: superchargingTokenConfig?.tokenSymbol })
+        ) : restrictSuperchargeForClaimOnly ? (
+          ''
```
Comment (jeanregisser) [inline src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx:307]: Shouldn't this condition be above hasMaxBalance first?

**Review by jeanregisser — APPROVED** [review]
Thanks!

File: src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx (hunk 1 above)
Comment (kathaypacific) [reply src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx:307]: the disclaimer for `hasMaxBalance` seems innocent enough which is why i didn't remove it, it currently doesn't encourage you to take any action (`"You’re getting maximum rewards on your {{token}}"`)

File: src/navigator/RewardsPill.tsx [hunk 2]
```diff
@@ -2,29 +2,39 @@ import React from 'react'
...
     ValoraAnalytics.track(RewardsEvents.rewards_screen_opened, {
       origin: RewardsScreenOrigin.RewardPill,
     })
   }
 
-  const rewardsEnabled = useSelector(rewardsEnabledSelector)
   const showRewardsPill = isE2EEnv || rewardsEnabled
 
-  if (!showRewardsPill) {
+  if ((restrictSuperchargeForClaimOnly && !isSupercharging) || !showRewardsPill) {
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/navigator/RewardsPill.tsx:37]: yes...my brain got real stuck on this this morning so i'll try again haha

[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/navigator/RewardsPill.tsx]: okay i think i did it, could you take a look? :D

Comment (bakoushin) [inline src/navigator/RewardsPill.tsx]: I would prefer to combine all conditions under `showRewardsPill` as its name suggests, but it is not a strong opinion.

```suggestion
  const isRestricted = restrictSuperchargeForClaimOnly && !isSupercharging
  const isEnabled = rewardsEnabled && !isRestricted
  const showRewardsPill =  isE2Env || isEnabled

  if (!showRewardsPill) {
     return null
  }
```

File: src/home/NotificationBox.tsx [hunk 3]
```diff
@@ -224,7 +230,7 @@ export function useSimpleActions() {
         })
       }
 
-      if (!isSupercharging && !dismissedStartSupercharging) {
+      if (!isSupercharging && !restrictSuperchargeForClaimOnly && !dismissedStartSupercharging) {
```
Comment (bakoushin) [inline src/home/NotificationBox.tsx:233]: We have a number of unit tests for supercharge notifications.
I wonder if we want to augment them for that case?

**Review by bakoushin — APPROVED** [review]

File: src/navigator/RewardsPill.tsx (hunk 2 above)
Comment (kathaypacific) [reply src/navigator/RewardsPill.tsx]: thanks, i think you commented on an outdated version of the code. what do you think of it now?

Comment (bakoushin) [reply src/navigator/RewardsPill.tsx]: Thanks! I think now it's much cleaner, all this complexity is in one place.

[the remark this answers is not in the document]
Comment (jeanregisser) [reply src/navigator/RewardsPill.tsx]: great all good! 👍

**Pull request merged**